cunumeric/eager.py

```diff
@@ -359,6 +359,20 @@ def flip(self, rhs, axes, stacklevel):
             self.array = np.flip(rhs.array, axes)
             self.runtime.profile_callsite(stacklevel + 1, False)
 
+    def choose(self, choices, out, mode, stacklevel):
+        if self.shadow:
+            choices = self.runtime.to_eager_array(
+                choices, stacklevel=(stacklevel + 1)
+            )
+            out = self.runtime.to_eager_array(out, stacklevel=(stacklevel + 1))
+        elif self.deferred is None:
+            self.check_eager_args((stacklevel + 1), choices, out)
+        if self.deferred is not None:
+            self.deferred.choose(choices, out, stacklevel=(stacklevel + 1))
+        else:
+            self.array[:] = np.choose(choices.array, out, mode)
```

Comment:
Should it not be `out.array` instead of `out`?

Reply:
Sure. Could you, please, help me to understand  what is eager array?

Reply:
Every cunumeric `ndarray` contains a thunk https://github.com/nv-legate/cunumeric/blob/branch-22.01/cunumeric/array.py#L199 that represents the computation which will produce the contents of that array. That thunk can either be an `EagerThunk` (this just performs the computation immediately on vanilla NumPy and stores it as `EagerThunk.array` https://github.com/nv-legate/cunumeric/blob/branch-22.01/cunumeric/eager.py#L34), or a `DeferredThunk` (this performs the computation asynchronously, by offloading it to Legion through a Legate Store https://github.com/nv-legate/cunumeric/blob/branch-22.01/cunumeric/array.py#L199).

Reply:
Thank you!